launcher/src/main/java/org/apache/spark/launcher/ChildProcAppHandle.java

```diff
@@ -171,10 +90,11 @@ void monitorChild() {
     }
 
     synchronized (this) {
-      if (disposed) {
+      if (isDisposed()) {
         return;
       }
 
+      State currState = getState();
```

Comment:
this was added just because `currState` is no longer accessible, right?  You're not particularly trying to grab the state before the call to `disconnect()`?  Might be clearly to move it after, otherwise on first read it looks like it is trying to grab the state before its modified by `disconnect()`